chronosQ-reservation-impl-redis/src/main/java/flipkart/cp/convert/chronosQ/impl/redis/RedisSchedulerStore.java

```diff
@@ -100,44 +103,52 @@ public Long remove(String value, long time, int partitionNum) throws SchedulerEx
     }
 
     @Override
-    public List<String> get(long time, int partitionNum) throws SchedulerException {
+    public List<SchedulerData> get(long time, int partitionNum) throws SchedulerException {
         Jedis jedis = null;
+        List<SchedulerData> schedulerDataList = new ArrayList<>();
         Set<String> resultSet;
         String key = "";
         try {
             jedis = _getInstance(partitionNum);
             key = getKey(time, partitionNum);
             resultSet = jedis.smembers(key);
             log.info("Get For " + key + "-" + resultSet);
+            for (String schedulerDataKey : resultSet) {
+                schedulerDataList.add(new SchedulerData(schedulerDataKey, jedis.get(schedulerDataKey)));
+            }
         } catch (Exception ex) {
             log.error("Exception occurred  for -" + "Key" + key + "Partition " + partitionNum + "-" + ex.getMessage());
             throw new SchedulerException(ex, ErrorCode.DATASTORE_READWRITE_ERROR);
         } finally {
             if ((jedis != null))
                 jedis.close();
         }
-        return new ArrayList<String>(resultSet);
+        return schedulerDataList;
     }
 
 
     @Override
-    public List<String> getNextN(long time, int partitionNum, int n) throws SchedulerException {
+    public List<SchedulerData> getNextN(long time, int partitionNum, int n) throws SchedulerException {
         Jedis jedis = null;
         List<String> resultSet;
+        List<SchedulerData> schedulerDataList = new ArrayList<>();
         String key = "";
         try {
             jedis = _getInstance(partitionNum);
             key = getKey(time, partitionNum);
             resultSet = jedis.srandmember(key, n);
             log.info("Get For " + key + "-" + resultSet);
+            for (String schedulerDataKey : resultSet) {
+                schedulerDataList.add(new SchedulerData(schedulerDataKey, jedis.get(schedulerDataKey)));
```

Comment:
A multi-get here would be more optimal here, since resultSet may contain a lot of entires.

Reply:
Replaced with mget.